internal/pkg/api/handleEnroll.go

```diff
@@ -197,10 +200,24 @@ func (et *EnrollerT) _enroll(
 ) (*EnrollResponse, error) {
 	var agent model.Agent
 	var enrollmentID string
+	var replaceToken string
 
 	span, ctx := apm.StartSpan(ctx, "enroll", "process")
 	defer span.End()
 
+	now := time.Now()
+
+	if req.ReplaceToken != nil && *req.ReplaceToken != "" {
+		replaceTokenBytes, err := bcrypt.GenerateFromPassword([]byte(*req.ReplaceToken), bcrypt.DefaultCost)
```

Comment:
I don't think the blowfish hash in the underlying implementation here meets FIPS 140-2. At least for the MS Go fork it wouldn't hit the FIPS certified OpenSSL so would be non-compliant. The list of what gets forwarded to it is at https://github.com/microsoft/go/blob/microsoft/main/eng/doc/fips/UserGuide.md

Whatever crypto primitives we use need to be FIPS compliant.

CC @michel-laterman

Reply:
That's actually more than a valid point @cmacknz 👍 AFAIK the only FIPS compliant password hashing is PBKDF2. I had a look and all call in it are forwarded by MS Go fork. We already use PBKDF2 password hashing in [elastic-agent](https://github.com/elastic/elastic-agent/blob/main/internal/pkg/crypto/io.go#L376) and I believe we should do the same here as well